### ISSUE TITLE
Dynamically link to org.kde.Platform

### ISSUE THREAD

**Issue by juxuanu** [flathub/org.telegram.desktop#324]
**Suggestion**: dynamically link to the KDE runtime (org.kde.Platform) instead of building statically.

**What'd be needed**: patches to Telegram-Desktop. Some distributions are (used?) to apply those, for example ArchLinux with its policy to try to avoid static linking when feasable. I can search for such patches if there is interest.

**Upstream had something to say about this 3 years ago**: https://github.com/telegramdesktop/tdesktop/issues/4688. The reason to not support dynamically linked builds of Telegram was 
>So that anyone could download a version from site and it just works.

But this is not an issue for Flatpak - you just depend on the `org.kde.Platform` and this is solved.

**Why bother?** See https://stackoverflow.com/a/1993407 

Would such patches be something of interest here?

**Comment by ilya-fedin** [comment]
org.kde.Platfrom was used previously. I disagree with the fact they're using Fedora's patch to enforce Wayland on GNOME (https://github.com/KDE/flatpak-kde-runtime/blob/qt5.15lts/patch/qtbase-use-wayland-on-gnome.patch), so I don't want to use it until they remove that patch, at least. But even if they do that, I'm not sure I want to replace Telegram's Qt patches that improve quality of life in a way with KDE's patchset that only backports merged to Qt changes.

TL;DR I don't think we need that.
